File: solutions/python/07.py
def draw_tree(height, ornament, frequency):
    tree = ""
    ornament_count = 0
    for i in range(1, height + 1):
        tree += " "*(height - i)
        for _ in range(0, 2 * (i - 1) + 1):
            ornament_count += 1
            if (ornament_count == frequency):
                tree += ornament
                ornament_count = 0
            else:
                tree += "*"
        tree += "\n"
    tree += " "*(height - 1) + "#"
    return tree


tree1 = draw_tree(5, "o", 2)
print(tree1)
#     *
#    o*o
#   *o*o*
#  o*o*o*o
# *o*o*o*o*
#     #

tree2 = draw_tree(3, "@", 3)
print(tree2)
#   *
#  *@*
# *@**@
#   #

tree3 = draw_tree(4, "+", 1)
print(tree3)
#    +
#   +++
#  +++++
# +++++++
#    #

"""
  Score: ★★★★★★★★ (3 bonus stars for solving it today!)
  Points: 340
  ops: 3787
"""

File: solutions/ts/07.ts
function drawTree(height: number, ornament: string, frequency: number): string {
  let tree = "";
  let ornamentCount = 0;
  for (let i = 1; i <= height; i++) {
    tree += " ".repeat(height - i);
    for (let j = 0; j < 2 * (i - 1) + 1; j++) {
      ornamentCount++;
      if (ornamentCount === frequency) {
        tree += ornament;
        ornamentCount = 0;
      } else {
        tree += "*";
      }
    }
    tree += "\n";
  }
  tree += " ".repeat(height - 1) + "#";
  return tree;
}

const tree1 = drawTree(5, "o", 2);
console.log(tree1);
//     *
//    o*o
//   *o*o*
//  o*o*o*o
// *o*o*o*o*
//     #

const tree2 = drawTree(3, "@", 3);
console.log(tree2);
//   *
//  *@*
// *@**@
//   #

const tree3 = drawTree(4, "+", 1);
console.log(tree3);
//    +
//   +++
//  +++++
// +++++++
//    #

/* 
  Score: ★★★★★★★★ (3 bonus stars for solving it today!)
  Points: 340
  ops: 4160
*/
